src/notebook/widget.ts

```diff
@@ -154,6 +154,16 @@ const TOOLBAR_PRESSED = 'jp-mod-pressed';
 const TOOLBAR_BUSY = 'jp-mod-busy';
 
 /**
+ * The class name added to a cell in edit mode.
```

Comment:
Now it seems clearer to me that the modes are *notebook* modes (like you call it in the model) and the class should be on the *notebook* widget. For example, we can't have some cells in edit mode and others in command mode. Instead, either the notebook as a whole is in edit mode (editing the current cell) or is in command mode.